src/main/java/org/opensearch/security/user/AuthCredentials.java

```diff
@@ -203,6 +217,14 @@ public Set<String> getBackendRoles() {
         return new HashSet<String>(backendRoles);
     }
 
+    /**
+     *
+     * @return Defensive copy of the security roles this user is member of.
+     */
+    public Set<String> getSecurityRoles() {
+        return Set.copyOf(securityRoles);
```

Comment:
Is `copyOf` needed here? The only place `addAll` is called is within the constructor

Reply:
It is for immutability, as by javadoc it states that this method returns a defensive copy.

Reply:
Moving the conversation here, since we want it to be immutable so I think we need this copyOf here.

Reply:
I see, I assumed it was for thread safety at first